src/application/models/v3/Appointments_model_v3.php

```diff
@@ -18,32 +18,44 @@
  */
 class Appointments_Model_V3 extends Appointments_Model {
 
-    public function getAppointmentWithIdUserIntegrated($id_user_integrated) {
+    public function getAppointmentsWithCondition($id_user_integrated, 
+                                                $id_service_integrated, 
+                                                $id_patient_integrated,
+                                                $startDate,
+                                                $endDate,
+                                                $page,
+                                                $size,
+                                                $sort) {     
+        $resultSet['total'] = $this->countAppointmentsByCondition($id_user_integrated, $id_service_integrated, $id_patient_integrated, $startDate, $endDate);
+        $arrayParams = $this->initStoredProcedureParams($id_user_integrated, $id_service_integrated, $id_patient_integrated, $startDate, $endDate);               
```

Comment:
extract code line 30 -> 34 to a method, like the way you have applied for $this->countAppointmentsByCondition(

Reply:
updated